### PR TITLE
Release: RST 4.7

### PR DESCRIPTION
**Pull request by ecbland**
# Release RST 4.7

Take two at making the release branch!

- [ ] PATCH
- [X] MINOR
- [ ] MAJOR

## Deadline: 11 April
Please comment in the Pull Request conversation if you need additional time to test the release.

## Testing

- [ ] Mac OSx
- [x] Linux deb - Debian, Ubuntu, Mint
- [ ] Linux rpm - OpenSuse, Fedora, CentOS

*Do any binaries/libraries require special attention during testing?*

Key updates in this release are
- New hardware file format (#455)
- Find frequency bands (#444)
- Data simulator updates (#436)
- Bug fixes for compilation on various systems
    - #450
    - #459
    - #462
- Anything else?

## Checklist

- [x] Update `radar.dat` (if required)
- [x] Update hardware files (if required)
- [x] Update `.rst.version`
- [x] Update author list in `.zenodo.json`

## Approving the release

After testing, please state which operating system you used and what you tested.

If you found no bugs during testing, please *approve* the PR:

1. Click on `files changed` top right
2. Click on `start review` green top right  button
3. In the comment box, state which operating system you used for testing, and any other helpful information
4. Select `approve` and `submit review`

### CODE REVIEW

**Review by mtwalach — APPROVED** [review]
Hi, I used git pull to update all the files in my RST directory and then reran the installation on a macOS Monterey v12.2 machine. 
As per usual, there were a few compilation warnings but it seems to have compiled just fine. This was my very very simple test: 

`rst walach$ make_grid --version
RST version: 4.7`

I haven't used it for anything else yet, but will be in touch if I run into any hurdles. 
🎉 Well done all of you!

**Comment by ecbland** [comment]
Based on our [authorship criteria](https://superdarn.github.io/dawg/documents/rst-authorship-criteria/), here is the proposed author list for RST 4.7:

_SuperDARN Data Analysis Working Group; Burrell, A.G.; Thomas, E.G.; Bland, E.C.; Coco, I.; Ponomarenko, P.V.; Schmidt, M.T.; Sterne, K.T.; Walach, M.-T._

We've identified the major contributors as: 
1. @aburrell, `find_freqband` tool (#444) 
2. @egthomas, hardware file format changes (#455), updates to data simulator (#436). 
 
Everyone else is listed alphabetically.

Please send any feedback to me & @ksterne, or leave a comment below. If anyone new helps with testing then of course we will add them to the author list, so I'll hold off updating the `.zenodo.json` file for now.

**Comment by egthomas** [comment]
That seems like quite the precipitous drop in the author list order for @mts299 who fixed a lot of compilation errors (which were correctly noted as a key update in the original message).

**Comment by ecbland** [comment]
> That seems like quite the precipitous drop in the author list order for @mts299 who fixed a lot of compilation errors (which were correctly noted as a key update in the original message).

Thanks @egthomas, I've updated the author list in my original post. Apologies to @mts299 for overlooking this.

**Comment by mts299** [comment]
Thank you @egthomas  and @ecbland. 

Going to try and install RST on our HPC this time around so this isn't an ongoing issue :p Will try to do this today.

**Comment by ksterne** [comment]
Has anyone been working with this on Ubuntu 20.04?  I pulled this release branch and have been getting a seg-fault on `make_fit` when I'm trying to run say:  `make_fit -fitacf-version 3.0 20150101.0000.02.bks.rawacf > bks_fitacf3` .  The seg-fault also occurs when I try 2.5 instead of 3.0 as well as no fitacf-version option.  Sorry I haven't been able to do any testing on this up until now.

**Comment by egthomas** [comment]
@ksterne that file processes correctly for me on this branch with both 2.5 and 3.0 (Ubuntu 18.04)

**Review by mts299 — COMMENTED** [review]
Runs on cedar and seems to be working fine.

HPC machine

File: codebase/superdarn/src.lib/tk/fitacf_v3.0/include/fit_structures.h [hunk 1]
```diff
@@ -106,6 +108,7 @@ typedef struct fit_prms {
   double **acfd;
   double **xcfd;
   int maxbeam;
+  double bmoff;
```
[the remark this answers is not in the document]
Comment (mtwalach) [reply codebase/superdarn/src.lib/tk/fitacf_v3.0/include/fit_structures.h:111]: Is it beam offset?

[the remark this answers is not in the document]
Comment (pasha-ponomarenko) [reply codebase/superdarn/src.lib/tk/fitacf_v3.0/include/fit_structures.h:111]: I believe this is a shift between the physical boresight direction (a perpendicular to the antenna array) and the centre of the field-of-view. 
"09) Electronic shift to radar scanning boresight (Degrees relative to
    physical antenna boresight. Normally 0.0 degrees)"
In the majority of radars these two directions coincide, but at some the field of view may be rotated by introducing additional linear phase delay along the array. E.g., at BKS this was implemented by connecting additional cables to each antenna.

File: README.md [hunk 2]
```diff
@@ -2,39 +2,52 @@
...
+
+Instructions for citing the RST are available on Zenodo. Click on DOI badge:
+[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.801458.svg)](https://doi.org/10.5281/zenodo.801458)
+Scroll to the bottom and copy the *Share Cite as* text or download the citation in your prefered format under *Export*.
+
+## Data access and usage policy
+
+SuperDARN data are published on [FRDR](https://www.frdr-dfdr.ca/repo/collection/superdarn) in the `rawacf`/`dat` format. These data are licensed under the [Creative Commons Attribution-NonCommercial 4.0 (CC BY-NC 4.0) license](https://creativecommons.org/licenses/by-nc/4.0/).
+
+**Please read the [SuperDARN data policy](https://g-772fa5.cd4fe.0ec8.data.globus.org/7/published/publication_285/submitted_data/2018RAWACF.readme.txt) when downloading data from FRDR.**
```
[the remark this answers is not in the document]
Comment (ecbland) [reply README.md:36]: @mts299 I would have liked to use a generic reference, but it doesn't exist, so I just chose the most recent one. Kathryn already recommended that we use the text from the readme files on FRDR, so this is the best we can do.

File: codebase/superdarn/src.lib/tk/fitacf_v3.0/src/fitacftoplevel.c [hunk 3]
```diff
@@ -166,9 +168,14 @@ void Copy_Fitting_Prms(struct RadarSite *radar_site, struct RadarParm *radar_prm
   fit_prms->interfer[0]=radar_site->interfer[0];
   fit_prms->interfer[1]=radar_site->interfer[1];
   fit_prms->interfer[2]=radar_site->interfer[2];
+  fit_prms->bmoff=radar_site->bmoff;
   fit_prms->bmsep=radar_site->bmsep;
   fit_prms->phidiff=radar_site->phidiff;
-  fit_prms->tdiff=radar_site->tdiff;
+  if ((radar_prms->offset == 0) || (radar_prms->channel < 2)) {
```
[the remark this answers is not in the document]
Comment (ecbland) [reply codebase/superdarn/src.lib/tk/fitacf_v3.0/src/fitacftoplevel.c:174]: @mts299 This looks like the same change that was proposed in #496. I think we need to hold off for now and let the TDIFF task force propose a solution.

**Comment by ecbland** [comment]
Hi All

Today was the deadline for testing the release branch. Has anyone done any testing that hasn't been reported here?

I've been testing on Linux Mint 20, compiles without any problems. 

I've also checked whether `make_fit` produces identical results on this branch and RST4.6. I used both versions of fitacf with the following files:
- `20150305.0601.00.inv.rawacf`
- `20150301.2001.00.cly.rawacf`
- `20180201.0001.00.lyr.rawacf`
- `20191026.0800.02.sas.1.rawacf`

The last file `20191026.0800.02.sas.1.rawacf` has different values of `elv`, `elv_low` and `elv_high`. This is due to changes in the beam separation for this radar (3.5 deg vs 3.24 deg)

File: codebase/superdarn/src.lib/tk/fitacf.2.5/src/fitacf.c [hunk 4]
```diff
@@ -78,7 +78,8 @@ struct FitBlock *FitACFMake(struct RadarSite *hd, int year) {
 }
 
 int fill_fit_block(struct RadarParm *prm, struct RawData *raw,
-                   struct FitBlock *input, struct FitData *fit){
+                   struct FitBlock *input, struct FitData *fit,
+                   struct RadarSite *hd){
```
[the remark this answers is not in the document]
Comment (egthomas) [reply codebase/superdarn/src.lib/tk/fitacf.2.5/src/fitacf.c:82]: Yes, this is the information from the hardware file and has been named following the same convention as in the existing `FitACFMake` and `FitACF` routines in `fitacf.c`, eg

https://github.com/SuperDARN/rst/blob/release/rst-4.7/codebase/superdarn/src.lib/tk/fitacf.2.5/include/fitacf.h#L29-L31

File: codebase/superdarn/src.lib/tk/elevation.1.0/include/elevation.h [hunk 5]
```diff
@@ -28,8 +28,8 @@ along with this program. If not, see <https://www.gnu.org/licenses/>.
 #ifndef _ELEVATION_H
 #define _ELEVATION_H
 
-double elevation_v2(struct FitPrm *prm, double psi_obs);
-double elevation(struct FitPrm *prm, double phi0);
-double elev_goose(struct FitPrm *prm, double phi0);
+double elevation(struct FitPrm * prm, double phi0);
+double elevation_v2(struct FitPrm * prm, double psi_obs);
+double elev_goose(struct FitPrm * prm, double phi0);
```
Comment (egthomas) [inline codebase/superdarn/src.lib/tk/elevation.1.0/include/elevation.h:33]:
```suggestion
double elevation(struct FitPrm *prm, double phi0);
double elevation_v2(struct FitPrm *prm, double psi_obs);
double elev_goose(struct FitPrm *prm, double phi0);
```

This seems like an unnecessary stylistic change from the rest of the RST.

File: codebase/superdarn/src.lib/tk/elevation.1.0/src/elev_goose.c [hunk 6]
```diff
@@ -20,17 +20,18 @@ You should have received a copy of the GNU General Public License
...
 
 #include <math.h>
 #include <stdio.h>
 
-#include "fitblk.h"
 #include "rmath.h"
+#include "fitblk.h"
 
-double elev_goose(struct FitPrm *prm, double phi0) {
+double elev_goose(struct FitPrm * prm, double phi0) {
```
Comment (egthomas) [inline codebase/superdarn/src.lib/tk/elevation.1.0/src/elev_goose.c:34]:
```suggestion
double elev_goose(struct FitPrm *prm, double phi0) {
```

This seems like an unnecessary stylistic change from the rest of the RST.

File: codebase/superdarn/src.lib/tk/elevation.1.0/src/makefile [hunk 7]
```diff
@@ -11,5 +11,6 @@ DSTPATH=$(LIBPATH)
 OUTPUT=elevation
 LINK="1"
 
+
```
Comment (egthomas) [inline codebase/superdarn/src.lib/tk/elevation.1.0/src/makefile:14]:
```suggestion
```

**Comment by ksterne** [comment]
A follow up on my previous comment where one machine I had was creating a seg fault on either fitting algorithm:  That particular machine is referencing the [SuperDARN/hdw ](https://github.com/SuperDARN/hdw) repo's hdw.dat files.  It may be worth making a point in the release notes that in order to use this new release, either the internal copy of the hdw.dat files should be used or the `NEW/format` branch of the SuperDARN/hdw repo.  Obviously I'm way behind on all this development!

I do remember noting that we were going ahead with the new hdw.dat format for this repo though.

**Comment by egthomas** [comment]
Can the `.travis.yml` file be removed?  I don't think we're going to start paying for the Travis-CI services anytime soon.

**Comment by ecbland** [comment]
I've created the draft release notes for RST 4.7 here: https://github.com/SuperDARN/rst/releases
Comments welcome. 

If nothing else comes up, I will merge this PR tomorrow (Wednesday)

**Comment by ksterne** [comment]
Should we add in a note that this release may or may not work with the old hardware file format but also note the new hardware file format is included within this release?

**Comment by ecbland** [comment]
> Should we add in a note that this release may or may not work with the old hardware file format but also note the new hardware file format is included within this release?

Good point...I've updated the draft release notes.

**Pull request merged**